packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -71,8 +71,8 @@ export interface ExportedDayCalendarProps<TDate> extends ExportedPickersDayProps
    */
   displayWeekNumber?: boolean;
   /**
-   * Calendar will show more weeks in order to match this value.
-   * Put it to 6 for having fix number of week in Gregorian calendars
+   * The calendar will show as many weeks as needed after the end of the current month to match this value.
```

Comment:
One alternative:

```suggestion
   * The day view will show as many weeks as needed after the end of the current month to match this value.
```

Reply:
How about we also add the following:

```
If not defined, the component will only contain the weeks that are part of the current month, meaning that the amount of weeks can vary from one month to another. 
```

Reply:
I like the idea of mentioning the `day view` instead of calendar 👍 
And I think the additional explanation makes a lot of sense 🎉

Reply:
Applied :+1: